tip-distributor/src/bin/claim-mev-tips.rs

```diff
@@ -16,37 +20,71 @@ struct Args {
     merkle_trees_path: PathBuf,
 
     /// RPC to send transactions through
-    #[arg(long, env)]
+    #[arg(long, env, default_value = "http://localhost:8899")]
     rpc_url: String,
 
     /// Tip distribution program ID
     #[arg(long, env)]
-    tip_distribution_program_id: String,
+    tip_distribution_program_id: Pubkey,
 
     /// Path to keypair
     #[arg(long, env)]
     keypair_path: PathBuf,
+
+    /// Number of unique connections to the RPC server for sending txns
+    #[arg(long, env, default_value_t = 100)]
+    rpc_send_connection_count: u64,
+
+    /// Rate-limits the maximum number of requests per RPC connection
+    #[arg(long, env, default_value_t = 200)]
+    max_concurrent_rpc_get_reqs: usize,
+
+    #[arg(long, env, default_value_t = 5)]
+    max_loop_retries: u64,
+
+    /// Limits how long before send loop runs before stopping. Defaults to 10 mins
+    #[arg(long, env, default_value_t = 10*60)]
+    max_loop_duration_secs: u64,
 }
 
-fn main() {
+#[tokio::main]
+async fn main() {
     env_logger::init();
-    info!("Starting to claim mev tips...");
-
     let args: Args = Args::parse();
+    info!("Starting to claim mev tips...");
+    let start = Instant::now();
 
-    let tip_distribution_program_id = Pubkey::from_str(&args.tip_distribution_program_id)
-        .expect("valid tip_distribution_program_id");
-
-    if let Err(e) = claim_mev_tips(
+    match claim_mev_tips(
         &args.merkle_trees_path,
-        &args.rpc_url,
-        &tip_distribution_program_id,
+        args.rpc_url,
+        args.rpc_send_connection_count,
+        args.max_concurrent_rpc_get_reqs,
+        &args.tip_distribution_program_id,
         &args.keypair_path,
-    ) {
-        panic!("error claiming mev tips: {:?}", e);
+        args.max_loop_retries,
+        Duration::from_secs(args.max_loop_duration_secs),
+    )
+    .await
+    {
+        Err(e) => datapoint_error!(
+            "claim_mev_workflow-claim_error",
+            ("error", 1, i64),
+            ("err_str", e.to_string(), String),
+            (
+                "merkle_trees_path",
+                args.merkle_trees_path.to_string_lossy(),
+                String
+            ),
+            ("latency_us", start.elapsed().as_micros(), i64),
```

Comment:
nit: elapsed_us

Reply:
zano usually suggests `latency_us`, hence been using that

Reply:
latency seems like wrong word here, but just a nit

Reply:
i've evolved from using latency for non-networking related measurements, that version of me is no longer present :)